lib/render/pxrUsdMayaGL/batchRenderer.cpp

```diff
@@ -1129,6 +1136,27 @@ UsdMayaGLBatchRenderer::_TestIntersection(
                  GL_TEXTURE_BIT |
                  GL_POLYGON_BIT);
 
+    // XXX: When Maya is using OpenGL Core Profile as the rendering engine (in
```

Comment:
To help me understand better, when is _TestIntersection() called?

Is UsdMayaGLBatchRenderer being called from the draw override solution from Pixar?

Reply:
Yeah, `UsdMayaGLBatchRenderer` is used to image `pxrUsdProxyShape` and `mayaUsdProxyShape` when `VP2_RENDER_DELEGATE_PROXY` is disabled.

The `MPxDrawOverride::userSelect()` override for proxy shapes is what ends up eventually calling `_TestIntersection()` to do viewport-based picking:
https://github.com/Autodesk/maya-usd/blob/c75e69638c560a82163f449cbcc3cbfbd8077e27/lib/render/pxrUsdMayaGL/proxyDrawOverride.cpp#L259